commit 3cd9bee6f053fb709ff1fcbff23ee5846dfdaf66
Author: Emilio Jesus Gallego Arias <e+git@x80.org>
Date:   Fri Dec 4 19:25:08 2015 +0100

    Replace Unix.readdir by Sys.readdir.
    
    This makes the function sightly more portable and has some advantages
    for some backends like the js_of_ocaml one.

diff --git a/lib/system.ml b/lib/system.ml
index 2e35a98..3aa8eeb 100644
--- a/lib/system.ml
+++ b/lib/system.ml
@@ -67,16 +67,7 @@ module StrSet = Set.Make(StrMod)
 let dirmap = ref StrMap.empty
 
 let make_dir_table dir =
-  let b = ref StrSet.empty in
-  let a = Unix.opendir dir in
-  (try
-    while true do
-      let s = Unix.readdir a in
-      if s.[0] != '.' then b := StrSet.add s !b
-    done
-  with
-  | End_of_file -> ());
-  Unix.closedir a; !b
+  Array.fold_left (fun s f -> StrSet.add f s) StrSet.empty (Sys.readdir dir)
 
 let exists_in_dir_respecting_case dir bf =
   let contents, cached =
